aspnetcore/tutorials/min-web-api.md

```diff
@@ -0,0 +1,341 @@
+---
+title: "Tutorial: Create a minimal web API with ASP.NET Core"
+author: rick-anderson
+description: Learn how to build a minimal web API with ASP.NET Core.
+ms.author: riande
+ms.date: 8/24/2021
+monikerRange: '>= aspnetcore-6.0'
+no-loc: [Home, Privacy, Kestrel, appsettings.json, "ASP.NET Core Identity", cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR, Models]
+uid: tutorials/min-web-api
+---
+
+<!-- test with windows sandbox -->
+# Tutorial: Create a web minimal API with ASP.NET Core
+
+By [Rick Anderson](https://twitter.com/RickAndMSFT)
+
+Minimal APIs are architected to create REST APIs with the minimal dependencies. They are ideal for microservice and app that don't want to include only the minimum files, features, and dependencies in ASP.NET Core.
+
+This tutorial teaches the basics of building a miminal web API with ASP.NET Core. For a tutorial on creating a web API project that contains more features, see [Create a web API](xref:tutorials/first-web-api)
+
+## Overview
+
+This tutorial creates the following API:
+
+|API | Description | Request body | Response body |
+|--- | ---- | ---- | ---- |
+|`GET /api` | Browser test, "Hello World" | None | `Hello World!`|
+|`GET /TodoItems` | Get all to-do items | None | Array of to-do items|
+|`GET /TodoItems/{id}` | Get an item by ID | None | To-do item|
+|`POST /TodoItems` | Add a new item | To-do item | To-do item |
+|`PUT /TodoItems/{id}` | Update an existing item &nbsp; | To-do item | None |
+|`DELETE /TodoItems/{id}` &nbsp; &nbsp; | Delete an item &nbsp; &nbsp; | None | None|
+
+## Prerequisites
+
+# [Visual Studio](#tab/visual-studio)
+
+[!INCLUDE[](~/includes/net-prereqs-vs-6.0.md)]
+
+# [Visual Studio Code](#tab/visual-studio-code)
+
+[!INCLUDE[](~/includes/net-prereqs-vsc-6.0.md)]
+
+# [Visual Studio for Mac](#tab/visual-studio-mac)
+
+[!INCLUDE[](~/includes/net-prereqs-mac-6.0.md)]
+
+---
+
+## Create an empty web project
+
+# [Visual Studio](#tab/visual-studio)
+
+* From the **File** menu, select **New** > **Project**.
+* Select the **ASP.NET Core Empty** template and select **Next**.
+* Name the project *TodoApi* and select **Next**.
+* In the **Additional information** dialog, select **.NET 6.0 (Preview)** and select **Create**.
+ 
+<!-- 
+![VS new project dialog](min-web-api/_static/5/vs.png)
+-->
+
+<!-- Move this later since we don't need it now -->
+# [Visual Studio Code](#tab/visual-studio-code)
+
+* Open the [integrated terminal](https://code.visualstudio.com/docs/editor/integrated-terminal).
+* Change directories (`cd`) to the folder that will contain the project folder.
+* Run the following commands:
+
+   ```dotnetcli
+   dotnet new webapi -o TodoApi
```

Comment:
This isn't the right template. This will add a weather forecast controller.

Reply:
This should be `dotnet new web`  not `dotnet new webapi `